src/Interpreters/evaluateConstantExpression.cpp

```diff
@@ -44,6 +44,9 @@ static std::pair<Field, std::shared_ptr<const IDataType>> getFieldAndDataTypeFro
 
 std::pair<Field, std::shared_ptr<const IDataType>> evaluateConstantExpression(const ASTPtr & node, const ContextPtr & context)
 {
+    if (!node->hasColumnName())
```

Comment:
Sorry, I do not fully understand the original problem. My question might be incorrect.

In scope of this 2 lines. Why if the is no column name then it is not not a constant expression. Is the reverse statement valid: node with column names is a constant expression?

Reply:
Usually, all expressions passed here have column names, but in some weird cases like `mysql('127.0.0.1:9004', currentDatabase(), 'foo', 'default', '', SETTINGS connection_pool_size = 1)` we may have unexpected AST type (ASTSet in the example). The table function can't be constant.

Another solution is to check the table function name and throw an error for `mysql` and others. Like here https://github.com/ClickHouse/ClickHouse/blob/a1397d7a0dc60d965249c74bfca4db3310f2a0ad/src/Parsers/ASTFunction.cpp#L474-L475

It may be less generic, though. So need to realize if there're any other cases.

Reply:
Thanks for clarification.
I understand now that we expect a constant expression as an argument. But encounter a function call.

But still I see some unclear statement: why presents of column names say anything about is expression constant or not? Furthermore my intuition says me an opposite thing, it is might be non constant if it has column name.